File: lit_cli/main/push.py
import click
import yaml
import requests
import toml
from pathlib import Path
from halo import Halo

CONFIG_PATH = Path("./main/config.yaml")
TOKEN_PATH = Path(".lit_env_data.toml")


# def load_config():
#     if CONFIG_PATH.exists():
#         with open(CONFIG_PATH, "r") as f:
#             return yaml.safe_load(f)
#     return {}
#


def load_data():
    if TOKEN_PATH:
        return toml.load(TOKEN_PATH)
    return None


def load_token():
    if TOKEN_PATH.exists():
        token = toml.load(TOKEN_PATH).get("auths", {}).get("token")
        return token
    return None


@click.command()
@click.option(
    "--file",
    "-f",
    type=click.Path(exists=True),
    help="File to push",
)
def push(file):
    token = load_token()
    API_URL = "http://localhost:8080"

    if not token:
        click.secho("🔑 Token not found. Please login first.", fg="red")
        return

    if not API_URL:
        click.secho("🌐 API URL not found in config.yaml.", fg="red")
        return

    file_path = Path(file)
    content = file_path.read_text()

    active_project = load_data().get("active_project", {})
    spinner = Halo(text="📤 Pushing data...", spinner="dots")
    spinner.start()

    try:
        response = requests.put(
            f"{API_URL}/projects/update-env-data/{active_project.get('id')}/",
            json={"envData": content},
            headers={"Authorization": f"Bearer {token}"},
            timeout=30,
        )

        if response.status_code == 200:
            spinner.succeed("✅ Push successful!")
        else:
            spinner.fail(f"❌ Push failed ({response.status_code})")
            click.secho(response.text, fg="red")

    except requests.RequestException as e:
        spinner.fail("🚨 Network error.")
        click.secho(str(e), fg="red")

File: lit_cli_v2/handlers/push.go
package handlers

import (
	"bytes"
	"encoding/json"
	"fmt"
	"io"
	"net/http"
	"os"
	"time"

	"github.com/briandowns/spinner"
	"github.com/pelletier/go-toml"
	"github.com/spf13/cobra"
)

const pushAPIURL = "http://localhost:8080"
const dataFilePath = ".lit_env_data.toml"

func loadPushToken() string {
	file, err := os.Open(dataFilePath)
	if err != nil {
		return ""
	}
	defer file.Close()

	var config map[string]map[string]string
	_ = toml.NewDecoder(file).Decode(&config)
	return config["auths"]["token"]
}

func loadActiveProjectID() string {
	file, err := os.Open(dataFilePath)
	if err != nil {
		return ""
	}
	defer file.Close()

	var config map[string]map[string]string
	_ = toml.NewDecoder(file).Decode(&config)
	return config["active_project"]["id"]
}

func PushEnvFile(filePath string) {
	token := loadPushToken()
	if token == "" {
		fmt.Println("🔑 Token not found. Please login first.")
		return
	}

	projectID := loadActiveProjectID()
	if projectID == "" {
		fmt.Println("📁 Active project not found. Please select a project first.")
		return
	}

	content, err := os.ReadFile(filePath)
	if err != nil {
		fmt.Println("❌ Failed to read file:", err)
		return
	}

	payload := map[string]string{"envData": string(content)}
	jsonBody, _ := json.Marshal(payload)

	s := spinner.New(spinner.CharSets[14], 100*time.Millisecond)
	s.Suffix = " 📤 Pushing data..."
	s.Start()

	req, _ := http.NewRequest("PUT", fmt.Sprintf("%s/projects/update-env-data/%s/", pushAPIURL, projectID), bytes.NewBuffer(jsonBody))
	req.Header.Set("Authorization", "Bearer "+token)
	req.Header.Set("Content-Type", "application/json")

	client := http.Client{Timeout: 30 * time.Second}
	res, err := client.Do(req)
	s.Stop()

	if err != nil {
		fmt.Println("🚨 Network error:", err)
		return
	}
	defer res.Body.Close()

	if res.StatusCode == 200 {
		fmt.Println("✅ Push successful!")
	} else {
		fmt.Printf("❌ Push failed (%d)\n", res.StatusCode)
		body, _ := io.ReadAll(res.Body)
		fmt.Println(string(body))
	}
}

var pushCmd = &cobra.Command{
	Use:   "push",
	Short: "Push .env file to remote project",
	Run: func(cmd *cobra.Command, args []string) {
		filePath, _ := cmd.Flags().GetString("file")
		if filePath == "" {
			cmd.Println("❗ Please specify a file using --file or -f")
			return
		}
		PushEnvFile(filePath)
	},
}

func init() {
	pushCmd.Flags().StringP("file", "f", "", "Path to the .env file to push")
	rootCmd.AddCommand(pushCmd)
}
